readthedocs/core/signals.py

```diff
@@ -49,6 +49,12 @@ def decide_if_cors(sender, request, **kwargs):  # pylint: disable=unused-argumen
     if request.path_info.startswith('/api/v2/sustainability'):
         return True
 
+    # Don't do domain checking for APIv2 when the Domain is known
+    if request.path_info.startswith('/api/v2/'):
+        domain = Domain.objects.filter(domain__icontains=host)
+        if domain.exists():
```

Comment:
Do we want to allow to access the API no matter which domain the request comes from?